### ISSUE TITLE
Deleted markers are still "annotated"

### ISSUE THREAD

**Issue by refmitchell** [DeepLabCut/DeepLabCut#774]
Not sure if this is a bug, hence problem.

**Environment details**
OS: Windows 10
DLC Version: 2.1.7
Anaconda env: DLC-CPU
Interface: GUI

**Problem**
Using the DLC GUI, Labeling ToolBox. When I place a label, remove it with middle-click and then attempt to re-place it again I get an error saying that the body part is already annotated.

**Steps to reproduce:**
In the LabelingToolbox interface:
1. Place label using right click
2. Delete label using middle-mouse click, confirm in the dialogue box.
3. Attempt to re-place label using right click

I get this error dialogue box:

![image](https://user-images.githubusercontent.com/10943932/86037290-0a7aa980-ba37-11ea-88fd-1ac82d1cd7bb.png)

There doesn't seem to be any related terminal output.

**Additional**
I can re-place the removed marker if I navigate away from the current frame and return (back one frame, then forward one). Maybe the frame should be refreshed on marker deletion?

**Comment by MMathisLab** [comment]
Yes, the idea is if you want to move the marker, drag it - if you delete it, as the pop up windows says, you cannot replace it. Why would you want to delete it and then put it back?

**Comment by refmitchell** [comment]
Thanks for the response.

On deletion, the confirmation window is:

![image](https://user-images.githubusercontent.com/10943932/86117770-8a505480-bac7-11ea-98aa-04be2ec0aa3f.png)

This window says nothing about points being irreplacable. 

The error window in my original issue says nothing about the point not being replacable; it says the point is "already annotated" despite the fact I removed the annotation (ergo, it wasn't annotated). It's the same error message which appears if you try to annotate the same point twice. Same error, two different (at least from the GUI level) scenarios.  

I understand where you're coming from in terms of "intended use". If a marker is removed why would it be added back? I have no clue; perhaps I deleted the wrong one, perhaps I decide upon deletion that the marker actually can be placed, etc. It's just weird behaviour as far as the interface is concerned. The error message is also unhelpful, if I've removed an annotation why does the error message tell me that the point is already annotated?

It's made worse by the fact that removing points is possible, you just have to navigate back one frame then forward and try again; or, if you saved you can reload from there and try again. It's just an interface problem, the functionality is already there. Why not just allow the user to do it without the workarounds?

**Comment by MMathisLab** [comment]
I understand your point, but the backend first is not straightforward and since this is an odd behavior for a user, it will have to be low priority to change this. Of course, we welcome contributions so if you want to edit the GUI /pandas interface for being sure it’s actually deleted, then please go ahead (that was the issue as it’s hard to remove what temp written, I e place holder now that you activated the point. We can delete the data which then drops the line once saved, but that required you to engage next/previous.)

I would also update to 2.1.8.2 as there was changes to the GUI (although I didn’t test this specific behavior). But once can now hit delete. 

Again, I wouldn’t recommend using the GUI this way in general, hence why me make it slow for you to delete (two clicks!).

So, in short, happy for a pull request!

**Comment by jeylau** [comment]
Hi @refmitchell! I have just pushed a fix; it will be part of the 2.2b7 release :)
